worker/uniter/relation/resolver.go

```diff
@@ -323,6 +323,11 @@ func (r *createdRelationsResolver) NextOp(
 		return nil, resolver.ErrNoOperation
 	}
 
+	// We don't need to do anything if there is a pending hook.
+	if localState.Kind == operation.RunHook && localState.Step == operation.Pending {
```

Comment:
Trying to look at the other examples, 'relationsResolver' uses:
```
        if localState.Kind != operation.Continue {
                return nil, resolver.ErrNoOperation
        }
```

I find it very strange that Kind is usually something like the hook being run, but here kind is a type of operation.

It just seems very unintuitive that the way to get a unit into an error is to return NoOperation.

Looking elsewhere,NoOperation tends to be surpressed. (if err != ErrNoOperation: return)

Reply:
Yes... I really really wish we could allocate some dev time to rewrite the uniter and resolvers from scratch... It took me 1h to track down why the error state gets set with the regular relation hook errors but was not for the relation-created even though you could see the errors splatted in the logs! Take a look for example at https://github.com/juju/juju/blob/2.8/worker/uniter/resolver.go#L107-L137

Reply:
Doesn't that just mean "if you get NoOpp try the next handler" not "report an error on the current thing because the thing failed"?